code/modules/medical/pathology/pathogen_benevolent.dm

```diff
@@ -349,3 +349,50 @@ datum/pathogeneffects/benevolent/neuronrestoration
 
 	may_react_to()
 		return "The pathogen appears to have a gland that may affect neural functions."
+
+datum/pathogeneffects/benevolent/sunglass
+	name = "Sunglass Glands"
+	desc = "The infected grew sunglass glands."
+	infect_type = INFECT_NONE
+	rarity = RARITY_UNCOMMON
+
+	proc/glasses(var/mob/living/carbon/human/M as mob)
+		var/obj/item/clothing/glasses/G = M.glasses
+		var/obj/item/clothing/glasses/N = new/obj/item/clothing/glasses/sunglasses()
+		M.show_message({"<span class='notice'>[pick("You feel cooler!", "You find yourself wearing sunglasses.", "A pair of sunglasses grow onto your face.")][G?" But you were already wearing glasses!":""]</span>"})
+		if (G)
+			N.loc = M.loc
+			var/turf/T = get_edge_target_turf(M, pick(alldirs))
+			N.throw_at(T,rand(0,5),1)
+		else
+			N.loc = M
+			N.layer = M.layer
+			N.master = M
+			M.glasses = N
+			M.update_clothing()
+
+	disease_act(var/mob/M as mob, var/datum/pathogen/origin)
+		if (!origin.symptomatic)
+			return
+		switch(origin.stage)
+			if (2 to 4)
+				if (ishuman(M))
+					if (!(M:glasses) || (!(istype(M:glasses, /obj/item/clothing/glasses/sunglasses)) && prob(50)))
+						if (prob(15))
+							glasses(M)
+			if (5)
+				if (ishuman(M))
+					if (!(M:glasses) || (!(istype(M:glasses, /obj/item/clothing/glasses/sunglasses)) && prob(50)))
+						if (prob(25))
+							glasses(M)
```

Comment:
Since you're doing the ishuman() anyways, might as well typecast instead of using :